lib/auth/auth.js

```diff
@@ -89,15 +89,26 @@ auth.generateV4Headers =
         const credentialScope =
             `${scopeDate}/${region}/${service}/aws4_request`;
         const timestamp = amzDate;
-        const query = request.query || {};
+        const query = request.query || data;
```

Comment:
Is it possible to have a request.query on the IAM requests?  Would it be clearer/safer to say that if the service is 'iam', use data as query?